astropy/stats/histogram.py

```diff
@@ -206,11 +205,9 @@ def freedman_bin_width(data, return_bins=False):
     dx = 2 * (v75 - v25) / (n ** (1 / 3))
 
     if return_bins:
+        if dx < 1e-6 * (v75 + v25):
```

Comment:
Hmm, looking at this again, this is obviously better, but still risky (what if v75 and v25 are both 0?
Multiple options:
1. Change to `<=`
2. Use `if np.isclose(v25, v75, rtol=1.e-6)`
3. Don't check `dx`, but rather check that `Nbins < 1e6` (say).

I think I like (3) best, as this is in fact what we're worried about.

Reply:
(3) is fine, but it will fail ungracefully in the case that inspired this PR (where dx = 0). That's why I suggested comparing dx to (dmax - dmin), which is effectively the same thing as looking at Nbins.

Reply:
Do I have to also check for `not np.isclose(dmax - dmin, 0)`?

Reply:
Ah, just saw your latest comment above. My browser not refreshing soon enough.

Reply:
I think `dx <= 1.e-6 * (dmax - dmin)` would cover everything (just need the `<=` for the pathelogical `dmax == dmin` case).